### PR TITLE
[net7.0] Update dependencies from xamarin/xamarin-macios to Xcode 14.3

### PR DESCRIPTION
**Pull request by dalexsoto**
Updates macios version to the ones we plan to ship as Xcode 14.3 support.

### CODE REVIEW

**Comment by dalexsoto** [comment]
@rmarinho FYI, this requires macOS Ventura

**Comment by rmarinho** [comment]
Ok we need some fixes

```
/Users/builder/azdo/_work/1/s/src/Core/src/ImageSources/iOS/ImageSourceExtensions.cs(21,49): error CS8604: Possible null reference argument for parameter 'attributes' in 'CGSize NSStringDrawing.GetSizeUsingAttributes(NSString This, UIStringAttributes attributes)'. [/Users/builder/azdo/_work/1/s/src/Core/src/Core.csproj::TargetFramework=net7.0-maccatalyst]
/Users/builder/azdo/_work/1/s/src/Core/src/Platform/iOS/LabelExtensions.cs(88,74): error CS8601: Possible null reference assignment. [/Users/builder/azdo/_work/1/s/src/Core/src/Core.csproj::TargetFramework=net7.0-maccatalyst]
/Users/builder/azdo/_work/1/s/src/Core/src/ImageSources/iOS/ImageSourceExtensions.cs(21,49): error CS8604: Possible null reference argument for parameter 'attributes' in 'CGSize NSStringDrawing.GetSizeUsingAttributes(NSString This, UIStringAttributes attributes)'. [/Users/builder/azdo/_work/1/s/src/Core/src/Core.csproj::TargetFramework=net7.0-ios]
/Users/builder/azdo/_work/1/s/src/Core/src/Platform/iOS/LabelExtensions.cs(88,74): error CS8601: Possible null reference assignment. [/Users/builder/azdo/_work/1/s/src/Core/src/Core.csproj::TargetFramework=net7.0-ios]
    36 Warning(s)
    4 Error(s)
```

File: src/Core/src/ImageSources/iOS/ImageSourceExtensions.cs [hunk 1]
```diff
@@ -18,18 +18,22 @@ public static partial class ImageSourceExtensions
 			var glyph = (NSString)imageSource.Glyph;
 
 			var attString = new NSAttributedString(glyph, font, color);
-			var imagesize = glyph.GetSizeUsingAttributes(attString.GetUIKitAttributes(0, out _));
+			var attributes = attString.GetUIKitAttributes(0, out _);
+			if (attributes != null)
```
Comment (rmarinho) [inline src/Core/src/ImageSources/iOS/ImageSourceExtensions.cs:22]: @mattleibow  any idea a better way here if this is null to get the imagesize  var below ?

**Comment by rmarinho** [comment]
@dalexsoto  I m hitting sometimes this  _manifest dependency 'Microsoft.NET.Workload.Emscripten.net7' version '7.0.4' is lower than version '7.0.5' required by manifest 'microsoft.net.workload.mono.toolchain.net7'_

```
dotnet-install: Installation finished successfully.
    Determining projects to restore...
    Restored /Users/builder/azdo/_work/2/s/src/DotNet/Dependencies/Workloads.csproj (in 1.17 sec).
  Unhandled exception: Microsoft.NET.Sdk.WorkloadManifestReader.WorkloadManifestCompositionException: Workload manifest dependency 'Microsoft.NET.Workload.Emscripten.net7' version '7.0.4' is lower than version '7.0.5' required by manifest 'microsoft.net.workload.mono.toolchain.net7' [/Users/builder/azdo/_work/2/s/bin/dotnet/sdk-manifests/7.0.200/microsoft.net.workload.mono.toolchain.net7/WorkloadManifest.json]
     at Microsoft.NET.Sdk.WorkloadManifestReader.WorkloadResolver.ComposeWorkloadManifests()
     at Microsoft.NET.Sdk.WorkloadManifestReader.WorkloadResolver.Create(IWorkloadManifestProvider manifestProvider, String dotnetRootPath, String sdkVersion, String userProfileDir)
     at Microsoft.DotNet.Workloads.Workload.InstallingWorkloadCommand..ctor(ParseResult parseResult, IReporter reporter, IWorkloadResolver workloadResolver, IInstaller workloadInstaller, INuGetPackageDownloader nugetPackageDownloader, IWorkloadManifestUpdater workloadManifestUpdater, String dotnetDir, String userProfileDir, String tempDirPath, String version, String installedFeatureBand)
     at Microsoft.DotNet.Workloads.Workload.Install.WorkloadInstallCommand..ctor(ParseResult parseResult, IReporter reporter, IWorkloadResolver workloadResolver, IInstaller workloadInstaller, INuGetPackageDownloader nugetPackageDownloader, IWorkloadManifestUpdater workloadManifestUpdater, String dotnetDir, String userProfileDir, String tempDirPath, String version, IReadOnlyCollection`1 workloadIds, String installedFeatureBand)
     at Microsoft.DotNet.Cli.WorkloadInstallCommandParser.<>c.<ConstructCommand>b__6_0(ParseResult parseResult)
     at System.CommandLine.Invocation.InvocationPipeline.<>c__DisplayClass4_0.<<BuildInvocationChain>b__0>d.MoveNext()
  --- End of stack trace from previous location ---
     at Microsoft.DotNet.Cli.Parser.<>c__DisplayClass17_0.<<UseParseErrorReporting>b__0>d.MoveNext()
  --- End of stack trace from previous location ---
     at System.CommandLine.CommandLineBuilderExtensions.<>c__DisplayClass11_0.<<UseHelp>b__0>d.MoveNext()
  --- End of stack trace from previous location ---
     at System.CommandLine.CommandLineBuilderExtensions.<>c.<<UseSuggestDirective>b__17_0>d.MoveNext()
  --- End of stack trace from previous location ---
     at System.CommandLine.CommandLineBuilderExtensions.<>c__DisplayClass15_0.<<UseParseDirective>b__0>d.MoveNext()
  --- End of stack trace from previous location ---
     at System.CommandLine.CommandLineBuilderExtensions.<>c__DisplayClass7_0.<<UseExceptionHandler>b__0>d.MoveNext()
/Users/builder/azdo/_work/2/s/src/DotNet/DotNet.csproj(170,5): error MSB3073: The command ""/Users/builder/azdo/_work/2/s/bin/dotnet/dotnet" workload install android --skip-sign-check --skip-manifest-update --verbosity diag --temp-dir "/Users/builder/azdo/_work/2/s/bin/temp/" --configfile "/Users/builder/azdo/_work/2/s/NuGet.config"" exited with code 1.

Build FAILED.

/Users/builder/azdo/_work/2/s/src/DotNet/DotNet.csproj(170,5): error MSB3073: The command ""/Users/builder/azdo/_work/2/s/bin/dotnet/dotnet" workload install android --skip-sign-check --skip-manifest-update --verbosity diag --temp-dir "/Users/builder/azdo/_work/2/s/bin/temp/" --configfile "/Users/builder/azdo/_work/2/s/NuGet.config"" exited with code 1.
    0 Warning(s)
    1 Error(s)
```

**Comment by rmarinho** [comment]
yeah bumping to 7.0.5 seems to fix it

**Review by rmarinho — APPROVED** [review]

**Comment by rmarinho** [comment]
@Redth @dalexsoto  we got the machines think in a good state, working on getting this merged.

**Comment by dalexsoto** [comment]
@rmarinho here is the net6.0 PR https://github.com/dotnet/maui/pull/14675

**Review by jsuarezruiz — APPROVED** [review]

**Pull request merged**